src/components/Fcrackzip/Fcrackzip.tsx

```diff
@@ -49,31 +69,32 @@ const Fcrackzip = () => {
             charSet: "",
         },
     });
-    // Ask per request of the tester, the checkzip will not disable the Setchar anymore but the logic will only be commented out
-    //if needed in the future we can uncomment the change back
-    // the useeffect is not really useful right now and maybe remove in future implementation
+
+    // Check for dependency availability when the component mounts
     useEffect(() => {
-        if (useCharsetLowercase || useCharsetUppercase || useCharsetNumeric) {
-            setCharsetSelected(true);
-        } else {
-            setCharsetSelected(false);
-        }
+        checkDependencyAvailability();
+    }, []);
 
-        //if (checkedUnzip) {
-        //setCharSetDisabled(true);
-        //} else {
-        //setCharSetDisabled(false);
-        //}
-    }, [useCharsetLowercase, useCharsetUppercase, useCharsetNumeric, checkedUnzip]);
+    // Function to check dependency availability
+    const checkDependencyAvailability = async () => {
+        const isAvailable = await checkAllCommandsAvailability(dependencies);
```

Comment:
This doesn't appear to be working in the app. Please test and review airbase-ng to see how the code should be implemented. 

To test, uninstall fcrackzip on your local machine, open DDT and navigate to the fcrackzip tool. If it is working, then the installation modal should open and ask you to install the dependency. 

It did not work for me after uninstalling fcrackzip.